sched/sched/sched_sysinfo.c

```diff
@@ -0,0 +1,100 @@
+/****************************************************************************
+ * sched/sched/sched_sysinfo.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/sysinfo.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <string.h>
+
+#include <nuttx/clock.h>
+#include <nuttx/kmalloc.h>
+#include <nuttx/pgalloc.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: sysinfo
+ *
+ * Description:
+ *   sysinfo() returns certain statistics on memory and swap usage,
+ *   as well as the load average.
+ *
+ ****************************************************************************/
+
+int sysinfo(FAR struct sysinfo *info)
+{
+#ifdef CONFIG_SCHED_CPULOAD
+  struct cpuload_s cpuload;
+#endif
+#ifdef CONFIG_MM_PGALLOC
+  struct pginfo_s pginfo;
+#endif
+  struct mallinfo minfo;
+
+  if (info == NULL)
+    {
+      set_errno(EINVAL);
+      return -1;
+    }
+
+  memset(info, 0, sizeof(*info));
+
+#ifdef CONFIG_SCHED_CPULOAD
+  clock_cpuload(0, &cpuload);
+
+  info->loads[0] = ((cpuload.total - cpuload.active) <<
+                     SI_LOAD_SHIFT) / cpuload.total;
+  info->loads[1] = info->loads[0];
+  info->loads[2] = info->loads[0];
```

Comment:
Should it have the logic for 5 and 15 minutes, case the current running time is more than 5 or more than 15 minutes?